pkg/reporter/reporter_test.go

```diff
@@ -51,12 +51,46 @@ var _ = PDescribe("Reporter", func() {
 		uploader      *RedHatInsightsUploader
 		generatedFile string
 
-		startStr = "2020-04-19T00:00:00Z"
+		startStr = "2020-07-18T00:00:00Z"
 		endStr   = "2020-07-19T00:00:00Z"
 		start, _ = time.Parse(time.RFC3339, startStr)
 		end, _   = time.Parse(time.RFC3339, endStr)
 	)
 
+	id := func(element interface{}) string {
+		return "row"
+	}
+
+	additionalLabels := Keys{
+		"namespace":             Equal("metering-example-operator"),
+		"pod":                   Equal("example-app-pod"),
+		"service":               Equal("example-app-pod"),
+		"persistentvolumeclaim": Equal("example-pvc"),
+	}
+
+	expectedMetricRow := Keys{
+		"additionalLabels":    MatchAllKeys(additionalLabels),
+		"domain":              Equal("apps.partner.metering.com"),
+		"interval_start":      HavePrefix("2020-"),
+		"interval_end":        HavePrefix("2020-"),
+		"kind":                Or(Equal("App"), Equal("App2")),
+		"metric_id":           BeAssignableToTypeOf(""),
+		"report_period_end":   Equal(endStr),
+		"report_period_start": Equal(startStr),
+		"rhmUsageMetrics": MatchAllKeys(Keys{
+			"rpc_durations_seconds_count": BeAssignableToTypeOf(""),
+			"rpc_durations_seconds_sum":   BeAssignableToTypeOf(""),
+		}),
+		"version": Equal(""),
+	}
+
+	expectedFields := Keys{
+		"report_slice_id": Not(BeEmpty()),
+		"metrics": MatchElements(id, AllowDuplicates, Elements{
+			"row": MatchAllKeys(expectedMetricRow),
+		}),
+	}
+
```

Comment:
This is a good idea - but each variables needs to be a var and they need to be set in a BeforeEach, if you don't bad things can happen like race conditions.

Test are randomly failing because of this.

Reply:
yep, will re-work